plugins/secrets/pages/SecretsPage.tsx

```diff
@@ -220,6 +243,16 @@ class SecretsPage extends mixin(StoreMixin) {
       selectedAction,
     } = this.state;
 
+    // the list-request will fail, thus this needs to be before the SecretsError-component
+    if (SecretStore.getStores().some((s) => s.sealed)) {
+      return (
+        <Page>
```

Comment:
moving this section up a couple lines is the fix. everything else is trimming down complexity (mainly removing structs in favor of static typing).